src/odemis/acq/stream/_projection.py

```diff
@@ -218,6 +223,437 @@ def _updateImage(self):
+                                 "azimuthal projection in reduced precision.",
+                                 data_bg_corr.shape)
+                    y, x = data_bg_corr.shape
+                    if y > x:
+                        small_shape = 1024, int(round(1024 * x / y))
+                    else:
+                        small_shape = int(round(1024 * y / x)), 1024
+                    # resize raw data
+                    data_bg_corr = img.rescale_hq(data_bg_corr, small_shape)
+
+                # define the size of the image for polar representation in GUI
+                # 2 x size of original/raw image (on smallest axis) and at most
+                # the size of a full-screen canvas (1134)
+                size = min(min(data_bg_corr.shape) * 2, 1134)
+
+                # TODO still valid or remove due to improved runtime of triangulation code??
+                # TODO: First compute quickly a low resolution and then
+                # compute a high resolution version.
+                # TODO: could use the size of the canvas that will display
+                # the image to save some computation time.
+
+                # Warning: allocates lot of memory, which will not be free'd until
+                # the current thread is terminated.
+
+                polar_data = angleres.AngleResolved2Polar(data_bg_corr, size, hole=False)
+
+                # TODO: don't hold too many of them in cache (eg, max 3 * 1134**2) - can this TODO be removed??
+                # TODO now we hold many more in cache
+                polar_dict[pos] = polar_data
+            except Exception:
+                logging.exception("Failed to convert to azimuthal projection")
+                return data  # display its raw as fallback
+
+        return polar_data
+
+    def _updateImage(self):
+        """Recomputes the image for the current ebeam position and polarization position requested."""
+        if not self.raw:
+            return
+
+        pos = self.stream.point.value
+        try:
+            if pos == (None, None):
+                self.image.value = None
+            else:
+                if self.stream._pos.keys()[0][-1]:
+                    pol = self.stream.polarization.value
+                else:
+                    pol = None
+                polar_data = self._project2Polar(pos + (pol,))
+
+                # update the histogram
+                # TODO: cache the histogram per image
+                # FIXME: histogram should not include the black pixels outside
+                # of the circle. => use a masked array?
+                # reset the drange to ensure that it doesn't depend on older data
+                self.stream._drange = None
+                self.stream._updateHistogram(polar_data)
+
+                self.image.value = self._project2RGB(polar_data)
+                # TODO why not store the RGB converted image in dict?? Because we need to calc the histogram
```

Comment:
Yes, because the main goal of caching the converted image is to make it fast when the user changes the brightness/contrast settings. If you were caching the image already projected, you'd still need to recompute it every time the B/C settings are changed.